### ISSUE TITLE
Semigroups breaking StructureDescription?

### ISSUE THREAD

**Issue by alex-konovalov** [gap-packages/Semigroups#393]
This happens with Semigroups 3.0.6 in the master branch of GAP:

```
-bash-4.2$ bin/gap.sh -r -A
 ┌───────┐   GAP 4.dev of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-gcc-default64
 Configuration:  gmp 6.0.0, readline
 Loading the library and packages ...
 Components: trans 1.0, prim 3.0, small* 1.0, id* 1.0
 Packages:   GAPDoc 1.6
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> LoadPackage("semi":OnlyNeeded);
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  IO 4.4.6 (Bindings for low level C library I/O routines)
by Max Neunhöffer (http://www-groups.mcs.st-and.ac.uk/~neunhoef).
Homepage: https://gap-packages.github.io/io
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  orb 4.7.6 (Methods to enumerate orbits)
by Juergen Mueller (http://www.math.rwth-aachen.de/~Juergen.Mueller),
   Max Neunhöffer (http://www-groups.mcs.st-and.ac.uk/~neunhoef), and
   Felix Noeske (http://www.math.rwth-aachen.de/~Felix.Noeske).
Homepage: https://gap-packages.github.io/orb
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  Digraphs 0.10.1 (Digraphs - Methods for digraphs)
by J. De Beule (http://homepages.vub.ac.be/~jdbeule/),
   J. Jonusas (http://www-groups.mcs.st-andrews.ac.uk/~julius/),
   J. D. Mitchell (http://goo.gl/ZtViV6),
   M. Torpey (http://www-groups.mcs.st-andrews.ac.uk/~mct25/), and
   W. A. Wilson (http://www-groups.mcs.st-andrews.ac.uk/~waw7/).
Homepage: https://gap-packages.github.io/Digraphs
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  genss 1.6.4 (Generic Schreier-Sims)
by Max Neunhöffer (http://www-groups.mcs.st-and.ac.uk/~neunhoef) and
   Felix Noeske (http://www.math.rwth-aachen.de/~Felix.Noeske).
Homepage: https://gap-packages.github.io/genss
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
-----------------------------------------------------------------------------
Loading  Semigroups 3.0.6
by J. D. Mitchell (http://www-groups.mcs.st-andrews.ac.uk/~jamesm/)
with contributions by:
     S. Burrell (http://sburrell.nfshost.com),
     M. Delgado (http://cmup.fc.up.pt/cmup/mdelgado/),
     J. East (http://goo.gl/MuiJu5),
     A. Egri-Nagy (http://www.egri-nagy.hu),
     N. Ham (https://n-ham.github.io),
     J. Jonusas (http://www-groups.mcs.st-andrews.ac.uk/~julius/),
     M. Pfeiffer (https://www.morphism.de/~markusp/),
     C. Russell,
     B. Steinberg (http://www.sci.ccny.cuny.edu/~benjamin/),
     J. Smith (http://math.sci.ccny.cuny.edu/people?name=Jhevon_Smith),
     M. Torpey (http://www-groups.mcs.st-and.ac.uk/~mct25/),
 and W. A. Wilson (http://wilf.me).
-----------------------------------------------------------------------------
true
gap> Test("tst/testinstall/opers/StructureDescription.tst");
########> Diff in tst/testinstall/opers/StructureDescription.tst:108
# Input is:
StructureDescription(G);
# Expected output:
"C0 x C0"
# But found:
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
The 1st argument is 'fail' which might point to an earlier problem
Error, no 1st choice method found for `Range' on 1 arguments
########
StructureDescription.tst
msecs: 3634
false
gap> 
```

**Comment by james-d-mitchell** [comment]
This one I can reproduce, just a method in Semigroups inappropriately overriding the default method in GAP.

**Comment by james-d-mitchell** [comment]
Resolved by PR #397
